subprojects/test-runner/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt

```diff
@@ -37,29 +37,18 @@ class VideoCapturerImpl(
     override fun start(): Result<Unit> {
         // checks if execute start() concurrently
         return if (state is State.Idling) {
-
-            val (videoFile, videoError) = testArtifactsProvider.generateUniqueFile("mp4")
-
-            if (videoError != null) {
-                Result.Failure(IllegalStateException("Can't create video file", videoError))
-            } else {
-                val (stdout, stdoutError) = testArtifactsProvider.generateFile("video-output", "txt", create = true)
-
-                if (stdoutError != null) {
-                    Result.Failure(IllegalStateException("Can't create video output file", videoError))
-                } else {
-                    try {
-                        executeRecorderCommand("start ${videoFile!!}", stdout!!)
-                        state = State.Recording(videoFile, stdout)
-                        Result.Success(Unit)
-                    } catch (t: Throwable) {
-                        if (videoFile!!.exists()) {
-                            videoFile.delete()
-                        }
-                        Result.Failure(IllegalStateException("Can't start video capturing", t))
-                    }
-                }
-            }
+            testArtifactsProvider.generateUniqueFile("mp4").flatMap(
```

Comment:
As for me in that code harder to make a bug because the compile check all nulls for you